swift/StableDiffusion/pipeline/StableDiffusionXLPipeline.swift

```diff
@@ -109,40 +146,30 @@ public struct StableDiffusionXLPipeline: StableDiffusionPipelineProtocol {
         progressHandler: (Progress) -> Bool = { _ in true }
     ) throws -> [CGImage?] {
 
-        // Encode the input prompt and negative prompt
-        let (promptEmbedding, pooled) = try encodePrompt(config.prompt)
-        let (negativePromptEmbedding, negativePooled) = try encodePrompt(config.negativePrompt)
+        // Determine input type of Unet
+        // SDXL Refiner has a latentTimeIdShape of [2, 5]
+        // SDXL Base has either [12] or [2, 6]
+        let isRefiner = unet.latentTimeIdShape.last == 5
+
+        // Setup geometry conditioning for base/refiner inputs
+        var baseInput: ModelInputs?
+        var refinerInput: ModelInputs?
+
+        // Check if the first textEncoder is available, which is required for base models
+        if textEncoder != nil {
+            baseInput = try generateConditioning(using: config, forRefiner: isRefiner)
+        }
+
+        // Check if the refiner unet exists, or if the current unet is a refiner
+        if unetRefiner != nil || isRefiner {
+            refinerInput = try generateConditioning(using: config, forRefiner: true)
```

Comment:
The `refinerInput` seems to have `1280` channels for the `hiddenStates` tensor while the `baseInput` has 2048. Is this expected? Asking because the code throws right after switching from base to refiner and trying to run the first forward pass on the refiner model:
```
multiArrayConstraint isAllowedValue: fails for MLFeatureTypeMultiArray feature encoder_hidden_states with featureValue MultiArray : Float32 2 × 1280 × 1 × 77 array. Error description: MultiArray shape (2 x 1280 x 1 x 77) does not match the shape (2 x 2048 x 1 x 77) specified in the model description
```

Reply:
This was due to an error on my part, please disregard (proper unetRefiner chunking will avoid this issue)